server-plugins/activity-resources/src/utils.ts

```diff
@@ -315,6 +323,15 @@ export async function getTxAttributesUpdates (
       }
     }
 
+    // we don't want to show collaborative documents in activity
+    // instead we show their content as Markup
+    // TODO this should be generalized via activity extension
+    if (attrClass === core.class.TypeCollaborativeDoc) {
```

Comment:
are you planning to do it  in this pull request or is it a note for the future?

Reply:
let's decide on review how we handle this
I can do in this PR but It would be also good to have async trigger because file loading may affect the trigger performance.

Reply:
we can make activity generation as asynс trigger, I thought about that too

I would prefer to do it async in a separate pull request, and merge this pull request for now if it works
You can assign this task to me

Reply:
I figured out that I need to do a bit more changes to support mixins here. Will be done tomorrow.